File: app/src/main/java/br/pedroso/upcomingmovies/MoviesApplication.java
package br.pedroso.upcomingmovies;


import android.app.Application;

import br.pedroso.upcomingmovies.di.ApplicationComponent;
import br.pedroso.upcomingmovies.di.ApplicationModule;
import br.pedroso.upcomingmovies.di.DaggerApplicationComponent;
import br.pedroso.upcomingmovies.network.di.NetworkModule;
import br.pedroso.upcomingmovies.repository.di.RepositoryModule;

public class MoviesApplication extends Application {

    private ApplicationComponent applicationComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        createApplicationComponent();
    }

    private void createApplicationComponent() {
        applicationComponent = DaggerApplicationComponent.builder()
                .applicationModule(new ApplicationModule(this))
                .networkModule(new NetworkModule(BuildConfig.MOVIES_DB_API_KEY))
                .repositoryModule(new RepositoryModule())
                .build();
    }

    public ApplicationComponent getApplicationComponent() {
        return applicationComponent;
    }

}

File: app/src/main/kotlin/br/pedroso/upcomingmovies/MoviesApplication.kt
package br.pedroso.upcomingmovies

import android.app.Application
import br.pedroso.upcomingmovies.di.ApplicationComponent
import br.pedroso.upcomingmovies.di.ApplicationModule
import br.pedroso.upcomingmovies.di.DaggerApplicationComponent
import br.pedroso.upcomingmovies.network.di.NetworkModule
import br.pedroso.upcomingmovies.repository.di.RepositoryModule


class MoviesApplication : Application() {
    lateinit var applicationComponent: ApplicationComponent
        private set

    override fun onCreate() {
        super.onCreate()

        createApplicationComponent()
    }

    private fun createApplicationComponent() {
        applicationComponent = DaggerApplicationComponent.builder()
            .applicationModule(ApplicationModule(this))
            .networkModule(NetworkModule(BuildConfig.MOVIES_DB_API_KEY))
            .repositoryModule(RepositoryModule())
            .build()
    }
}
